src/api/poll.ts

```diff
@@ -0,0 +1,85 @@
+import Bugsnag from "@bugsnag/js";
+import noop from "lodash/noop";
+
+import { PollMessage, PollValues } from "types/chat";
+
+import { getVenueRef } from "./venue";
+
+export interface CreatePollProps {
+  venueId: string;
+  poll: PollMessage;
+}
+
+export const createVenuePoll = async ({
+  venueId,
+  poll,
+}: CreatePollProps): Promise<void> =>
+  getVenueRef(venueId)
+    .collection("chats")
+    .add(poll)
+    .then(noop)
+    .catch((err) => {
+      Bugsnag.notify(err, (event) => {
+        event.addMetadata("context", {
+          location: "api/chat::createVenuePoll",
+          venueId,
+          poll,
+        });
+      });
+      // @debt rethrow error, when we can handle it to show UI error
+    });
+
+export type DeleteVenuePollProps = {
+  venueId: string;
+  pollId: string;
+};
+
+// RESTRICRED due to user rights issue
+export const deleteVenuePoll = async ({
+  venueId,
+  pollId,
+}: DeleteVenuePollProps): Promise<void> =>
+  getVenueRef(venueId)
+    .collection("chats")
+    .doc(pollId)
+    .update({ deleted: true })
+    .catch((err) => {
+      Bugsnag.notify(err, (event) => {
+        event.addMetadata("context", {
+          location: "api/chat::deleteVenuePoll",
+          venueId,
+          pollId,
+        });
+      });
+      // @debt rethrow error, when we can handle it to show UI error
+    });
+
+export type VoteInPollProps = {
+  venueId: string;
+  poll: PollValues;
+  votes: string[];
+  pollId: string;
+};
+
+// RESTRICRED due to user rights issue
+export const voteInVenuePoll = async ({
+  venueId,
+  poll,
+  votes,
+  pollId,
+}: VoteInPollProps): Promise<void> =>
+  getVenueRef(venueId)
+    .collection("chats")
+    .doc(pollId)
+    .update({ votes, poll })
+    .catch((err) => {
+      Bugsnag.notify(err, (event) => {
+        event.addMetadata("context", {
+          location: "api/chat::voteInVenuePoll",
+          venueId,
+          poll,
+          pollId,
+        });
+      });
+      // @debt rethrow error, when we can handle it to show UI error
+    });
```

Comment:
IMO voting should be implemented as a backend function. You must also store the voted user ids. In that backend you'll get the context of the caller, check if a person has voted. If not, add one vote.

Reply:
@mike-lvov haven’t looked too deeply into it, but sounds like a reasonable idea to me.

Reply:
fixed in 4e7f06c35c6b2ca0f88795306c915c2bf8143e3a